src/components/Pagination.tsx

```diff
@@ -57,7 +38,19 @@ const Pagination: React.FC<PaginationProps> = ({
           }`}>
           <button onClick={handlePrevPage}>&#8592;</button>
         </li>
-        {renderPageNumbers()}
+        {pages.map((p, i) => (
+          <button onClick={() => handlePageChange(i + 1)}>
+            <li
+              key={i}
+              className={`m-2 p-2 shadow-sm rounded ${
+                currentPage === i + 1
+                  ? 'text-gray-800 bg-lime-300 hover:text-white hover:bg-red-900'
+                  : 'text-white bg-red-900  hover:text-gray-800 hover:bg-lime-300'
+              }`}>
+              {i + 1}
+            </li>
+          </button>
+        ))}
```

Comment:
It's kind of an antipattern to use array.push for elements on the page and instead use .map or similar which will create a new array on render and its contents will be html elements that react will render